R/bs.R

```diff
@@ -168,29 +168,20 @@ prep.step_bs <- function(x, training, info = NULL, ...) {
 
 #' @export
 bake.step_bs <- function(object, new_data, ...) {
-  check_new_data(names(object$objects), object, new_data)
-
-  ## pre-allocate a matrix for the basis functions.
-  new_cols <- vapply(object$objects, ncol, c(int = 1L))
-  bs_values <-
-    matrix(NA, nrow = nrow(new_data), ncol = sum(new_cols))
-  colnames(bs_values) <- rep("", sum(new_cols))
-  strt <- 1
-  for (i in names(object$objects)) {
-    cols <- (strt):(strt + new_cols[i] - 1)
-    orig_var <- attr(object$objects[[i]], "var")
-    bs_values[, cols] <-
-      bs_predict(object$objects[[i]], new_data[[i]])
-    new_names <-
-      paste(orig_var, "bs", names0(new_cols[i], ""), sep = "_")
-    colnames(bs_values)[cols] <- new_names
-    strt <- max(cols) + 1
-    new_data <- remove_original_cols(new_data, object, orig_var)
+  col_names <- names(object$objects)
+  check_new_data(col_names, object, new_data)
+
+  for (col_name in col_names) {
+    new_values <- bs_predict(object$objects[[col_name]], new_data[[col_name]])
+    new_values <- as_tibble(new_values)
```

Comment:
I'm a bit hesitant about bringing `as_tibble()` inside of this loop, with an eye for performance. I tried pulling it out, as 

``` r
new_data <- as_tibble(new_data)
```

and it seemed to work fine. Any thoughts on this?

Reply:
I guess this is fine, the reason I did it because I wasn't expecting `vec_cbind()` to work with matrix input, but it appears to work well.

``` r
vctrs::vec_cbind(
  tibble::tibble(x = 2), 
  matrix(1, dimnames = list(NULL, "y"))
)
#> # A tibble: 1 × 2
#>       x     y
#>   <dbl> <dbl>
#> 1     2     1
```

In that case we wouldn't need `as_tibble()` at all. Because is guaranteed  that `new_data` will be a tibble